client/scala/src/main/scala/ai/verta/repository/Commit.scala

```diff
@@ -21,4 +23,45 @@ class Commit(
     case other: Commit => commit.commit_sha.get == other.commit.commit_sha.get
     case _ => false
   }
+
+  /** Creates a branch at this Commit and returns the checked-out branch
+   *  If the branch already exists, it will be moved to this commit.
+   *  @param branch branch name
+   *  @return if not saved, a failure; otherwise, this commit as the head of `branch`
+   */
+  def newBranch(branch: String)(implicit ec: ExecutionContext) = {
+    if (!saved)
+      Failure(new IllegalStateException("Commit must be saved before it can be attached to a branch"))
+    else setBranch(branch)
+  }
+
+  /** Assigns a tag to this Commit
+   *  @param tag tag
+   */
+  def tag(tag: String)(implicit ec: ExecutionContext) = {
+    if (!saved)
+      Failure(new IllegalStateException("Commit must be saved before it can be tagged"))
+    else clientSet.versioningService.SetTag2(
+        body = "\"" + commit.commit_sha.get + "\"",
+        repository_id_repo_id = repo.id.get,
+        tag = URLUtils.urlEncode(tag)
+    ).map(_ => ())
+  }
+
+  /** Set the commit of named branch to current commit
+   *  @param branch branch
+   */
+  private def setBranch(branch: String)(implicit ec: ExecutionContext) = {
+    clientSet.versioningService.SetBranch2(
+      body = "\"" + commit.commit_sha.get + "\"",
```

Comment:
Use Scala formatting (either inline or not) instead to make this more readable.

Reply:
You mean the string formatting/interpolation? I think previously I was having some issue because mixing that will the double quotes is kinda buggy. I'll give it another try.

Reply:
Is this what you're referring to:
https://github.com/scala/bug/issues/6476
It's a really long conversation, but the issue is still open at the end.

Reply:
Hm. Let me put in another way. If you send a string to `body`, it should serialize as a string in json, right? You shouldn't have to add the extra quotes. But why doesn't it work?

Reply:
Hmm originally I didn't put the quotations in either. It gave me a really weird JSON-parsing error. Let me try it again and send the error message here.

Reply:
The error object return is this:
ai.verta.swagger.client.HttpException: [400] {"error":"json: cannot unmarshal number into Go value of type string","code":3,"message":"json: cannot unmarshal number into Go value of type string"}

Reply:
Ok I managed to use put together an utility function based on net.lift.json's method. A bit less hacky. Resolved.

Reply:
It's isn't much less hacky. Take a moment to think about your solution. Do you think it's a good idea to have the person adding a new functionality to have to think whether their particular case will fall into the class of issues you faced here and whether they need to call this method? Or is there a way that you can hide this complexity under the hood and automatically account for all such scenarios?

Reply:
Oh I see what you mean. To avoid having to deal with this for new functionality, I'll have to put it where the query parameters are processed. I'm not quite sure if this hack is required at any other place though (I don't recall having to added double quotes to commit sha when I implemented other methods, only in this particular place), so a blanket fix (apply the method to all incoming parameters) is dangerous. I'll think more about this.

Reply:
Ok so this is what I did: I move the jsonFormat inside the request method of HttpClient instead. This way, when I implement a new functionality involving the passing of a body, it will be automatically handled, instead of me having to manually apply the function again. Does it sound any better?

Reply:
BTW I've also moved the jsonFormat function inside HttpClient, we're not gonna use it outside of this context.